### PR TITLE
csv downloader

### PR DESCRIPTION
**Pull request by YontiLevin**
the downloaded file still needs more formatting
but i want to merge so i wouldn't be too many commits behind

### CODE REVIEW

File: app/search/search.component.css [hunk 1]
```diff
@@ -1,16 +1,1133 @@
...
-}
-
-.search-icon {
-  cursor: pointer;
-}
-
-a.activeclass {
-    color: #F00 !important;
-}
+@import url("https://fonts.googleapis.com/css?family=Miriam+Libre");
```
Comment (OriHoch) [inline app/search/search.component.css]: are all these css changes intentional?

File: app/search/search.component.css.map [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (OriHoch) [inline app/search/search.component.css.map]: I don't think this file should be here

File: app/search/search.component.html [hunk 3]
```diff
@@ -11,10 +11,13 @@
                (keyup)="search(searchBox.value)"
                [value]="term"
                />
+        <span class="input-group-addon downloader">
+            <img class="downloader" (click)="download(term)" src='../../assets/img/download-csv.svg' />
```
Comment (OriHoch) [inline app/search/search.component.html]: I think it's better to include images from css, like this - 
https://github.com/OpenBudget/budgetkey-app-search/blob/master/app/styles.scss#L637

if you have to do it from the img tag, then I would check if referring to the img with ../../ is right, I think it's dependant on the URL of the page, which might be problematic. Worth to check what's the angular way of doing it (I'm not sure myself..)

File: app/search/search.component.ts [hunk 4]
```diff
@@ -113,6 +117,11 @@ export class SearchComponent implements OnInit {
     }
   }
 
+  download(term: string): void {
```
Comment (OriHoch) [inline app/search/search.component.ts]: looks like `term` parameter is not used?

File: app/search/search.component.css (hunk 1 above)
Comment (YontiLevin) [reply app/search/search.component.css]: no

File: app/search/search.component.css.map (hunk 2 above)
Comment (YontiLevin) [reply app/search/search.component.css.map]: true
i'll add it to the gitignore

File: app/search/search.component.html (hunk 3 above)
Comment (YontiLevin) [reply app/search/search.component.html]: i'll check it out

File: app/search/search.component.ts (hunk 4 above)
Comment (YontiLevin) [reply app/search/search.component.ts]: true
i should add it to the csv somewhere

File: app/_service/download.service.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    console.log(unique);
+
+    for (let key of unique){
+      let headers = headers4type[key];
+      csvFile += addTitle(key, headers);
+      for (let entry of allDocs.value){
+        if (entry.type === key) {
+          csvFile += key;
+          for (let h of headers){
+            csvFile += ',' + entry.source[h];
```
[the remark this answers is not in the document]
Comment (YontiLevin) [reply app/_service/download.service.ts]: i'll check it out

File: app/search/search.component.css (hunk 1 above)
Comment (akariv) [reply app/search/search.component.css]: so let's revert :)

**Comment by YontiLevin** [comment]
fixed your comments
still more work is needed on the file formatting

File: app/_service/download.service.ts [hunk 6]
```diff
@@ -55,23 +42,25 @@ export class DownloadService {
 
     for (let key of unique){
       let headers = headers4type[key];
-      csvFile += addTitle(key, headers);
+      csvFile += Papa.unparse(Array(headers));
```
Comment (akariv) [inline app/_service/download.service.ts]: Have you tried using the PapaParse.unparse method variant that gets the list of fields and the data as an array of objects?
Might be easier to use that this loop.

Comment (akariv) [reply app/_service/download.service.ts]: Either way I think that rewriting this using the `map` method would make the code much more readable.

**Comment by YontiLevin** [comment]
- [x] Papa.unparse({fields:[],data:[[]]})

 **X** map instead of nested fors

**Pull request merged**